lib/core/topologies/replset_state.js

```diff
@@ -420,6 +420,8 @@ ReplSetState.prototype.update = function(server) {
   //
   if (!this.primary && ismaster.ismaster && ismaster.setName) {
     var ismasterElectionId = server.lastIsMaster().electionId;
+    this.setName = this.setName.trim();
+    ismaster.setName = ismaster.setName.trim();
```

Comment:
Why are we modifying `ismaster`?

Reply:
fixed this so that the if statement compares the trimmed setNames, but the variables themselves are not modified